src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -150,14 +157,32 @@ private String resolveHeadSha() {
                             getRepository(), head.get().getName()));
         }
 
-        if (revision.get() instanceof AbstractGitSCMSource.SCMRevisionImpl) {
-            return ((AbstractGitSCMSource.SCMRevisionImpl) revision.get()).getHash();
+        return resolveCommit(revision.get());
+    }
+
+    private String resolveCommit() {
+        if (run == null) {
+            return StringUtils.EMPTY;
```

Comment:
returning an empty string is usually a codesmell, just return null if you can't find it, imo
```suggestion
            return null;
```

Reply:
Then we need to guard all methods that invoke the `getSha`.

Reply:
This method is only called internally from one place?